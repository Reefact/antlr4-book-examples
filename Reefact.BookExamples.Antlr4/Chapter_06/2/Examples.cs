#region Usings declarations

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_06._2 {

    [UseReporter(typeof(BeyondCompareReporter))]
    public class Examples {

        [Fact]
        public void t_json_tokensString() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("t.json", 6, 2);
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string tokensString = grun.ToTokensString();
            // Verify
            Approvals.Verify(tokensString);
        }

        [Fact]
        [GraphicalTree("ParseTree1.svg")]
        public void t_json_mermaid_style_tree() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("t.json", 6, 2);
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        public void simple_array_tokens() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("[1,\"\\u0049\",1.3e9]");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string tokensString = grun.ToTokensString();
            // Verify
            Approvals.Verify(tokensString);
        }

        [Fact]
        public void simple_array_tree() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("[1,\"\\u0049\",1.3e9]");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string lispStyleTree = grun.ToLispStyleTree();
            // Verify
            Approvals.Verify(lispStyleTree);
        }

    }

}