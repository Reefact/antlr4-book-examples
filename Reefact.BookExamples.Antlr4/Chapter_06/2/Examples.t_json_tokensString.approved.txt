[@0,0:0='{',<1>,1:0]
[@1,5:15='"antlr.org"',<10>,2:2]
[@2,16:16=':',<4>,2:13]
[@3,18:18='{',<1>,2:15]
[@4,25:32='"owners"',<10>,3:4]
[@5,33:33=':',<4>,3:12]
[@6,35:35='[',<5>,3:14]
[@7,36:36=']',<6>,3:15]
[@8,37:37=',',<2>,3:16]
[@9,44:49='"live"',<10>,4:4]
[@10,50:50=':',<4>,4:10]
[@11,52:55='true',<7>,4:12]
[@12,56:56=',',<2>,4:16]
[@13,63:69='"speed"',<10>,5:4]
[@14,70:70=':',<4>,5:11]
[@15,72:76='1e100',<11>,5:13]
[@16,77:77=',',<2>,5:18]
[@17,84:90='"menus"',<10>,6:4]
[@18,91:91=':',<4>,6:11]
[@19,93:93='[',<5>,6:13]
[@20,102:107='"File"',<10>,7:6]
[@21,108:108=',',<2>,7:12]
[@22,117:128='"Help\nMenu"',<10>,8:6]
[@23,135:135=']',<6>,9:4]
[@24,141:141='}',<3>,10:2]
[@25,145:145='}',<3>,11:0]
[@26,148:147='<EOF>',<-1>,12:0]